site/topics/logic/truth-tables.rst

```diff
@@ -2,14 +2,91 @@
+
 
 
 Boolean Logic
 =============
 
+* Boolean logic is a form of algebra that operates on boolean values that can take on only two states
+* These states are typically called *True* and *False*
 
-Boolean Algebra
----------------
+    * Depending on context, these are sometimes referred to as *On*/*Off*, ``1``/``0``, or *high voltage*/*low voltage*
+
+
+* Within the algebra, the *operands* are values that take on one of these two states
+* The *operators* act on these operands to produce a new boolean value
+
+
+.. note::
+
+    If this terminology is throwing you off, remember that this is like the integer operators/operands you are familiar
+    with. Consider the expression :math:`1 + 2`. The operands here are integers :math:`1` and :math:`2` and the operator
+    is :math:`+`, which means addition. Here, addition is an operator that takes two integer values and produces a new
+    integer value.
+
+    With boolean logic, the operands can have one of two boolean values as opposed to an infinite number, like with
+    integers, and the operators will take two boolean values to produce a new boolean value.
+
+
+
+* There are three basic boolean operators
+
+    * **Not**
+
+        * Unary operator --- only operates on a single operand to produce a single value
+        * Given some boolean value :math:`a`, invert it
+
+            * In other words, if :math:`a` is *True*, not :math:`a` is *False*, and *vice versa*
+
+
+        * Typically denoted as :math:`\lnot a` or sometimes :math:`\overline a`
+
+
+    * **And**
+
+        * Binary operator --- operates on two operands to produce a single value
+        * Given boolean values :math:`a` and :math:`b`, return *True* if both values are *True*, *False* otherwise
+        * Denoted as :math:`a \land b`
+
+
+    * **Or**
+
+        * Binary operator --- operates on two operands to produce a single value
+        * Given boolean values :math:`a` and :math:`b`, return *True* if both or either are *True*, *False* otherwise
+        * Denoted as :math:`a \lor b`
+
+
+* There are additional boolean operators that can be made up from the three basic operators
+* Three of these are commonly used within the context of computer architecture, thus they will be presented here
+
+    * **Exclusive Or** (**XOR**)
+
+        * Binary operator --- operates on two operands to produce a single value
+        * Given boolean values :math:`a` and :math:`b`, return *True* if and only if either are *True*, *False* otherwise
+
+            * Similar to or, but if both are *True*, it returns *False*
+
+        * Denoted as :math:`a \oplus b`
+        * Equivalent to :math:`(a \lor b) \land \lnot (a \land b)`
+
+
+    * **Not Or** (**NOR**)
+
+        * Literally *not or*
+        * Sometimes denoted as :math:`\overline \lor`
+        * Equivalent to :math:`\lnot (a \lor b)`
+        * Functionally complete --- can be used to generate all other boolean operators
```

Comment:
I actually didn't know that about nor. I thought that was a nand thing only.

Reply:
I legit found out about 1.hour ago.